src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -956,119 +957,150 @@ private static void ExpandTypeName(TypeSyntax type, StringBuilder builder)
         {
-            string name;
-            TypeParameterListSyntax typeParameterList;
+            const string missingTokenPlaceholder = "?";
+
             switch (node.Kind())
             {
-                case SyntaxKind.ClassDeclaration:
-                    var classDecl = (ClassDeclarationSyntax)node;
-                    name = classDecl.Identifier.ValueText;
-                    typeParameterList = classDecl.TypeParameterList;
-                    break;
                 case SyntaxKind.CompilationUnit:
-                    return string.Empty;
-                case SyntaxKind.DelegateDeclaration:
-                    var delegateDecl = (DelegateDeclarationSyntax)node;
-                    name = delegateDecl.Identifier.ValueText;
-                    typeParameterList = delegateDecl.TypeParameterList;
-                    break;
-                case SyntaxKind.EnumDeclaration:
-                    return ((EnumDeclarationSyntax)node).Identifier.ValueText;
+                    return null;
                 case SyntaxKind.IdentifierName:
-                    return ((IdentifierNameSyntax)node).Identifier.ValueText;
-                case SyntaxKind.InterfaceDeclaration:
-                    var interfaceDecl = (InterfaceDeclarationSyntax)node;
-                    name = interfaceDecl.Identifier.ValueText;
-                    typeParameterList = interfaceDecl.TypeParameterList;
-                    break;
-                case SyntaxKind.MethodDeclaration:
-                    var methodDecl = (MethodDeclarationSyntax)node;
-                    name = methodDecl.Identifier.ValueText;
-                    typeParameterList = methodDecl.TypeParameterList;
-                    break;
+                    var identifier = ((IdentifierNameSyntax)node).Identifier;
+                    return identifier.IsMissing ? missingTokenPlaceholder : identifier.Text;
                 case SyntaxKind.NamespaceDeclaration:
-                    return GetNodeName(((NamespaceDeclarationSyntax)node).Name, includeTypeParameters: false);
+                    return GetName(((NamespaceDeclarationSyntax)node).Name, options);
                 case SyntaxKind.QualifiedName:
                     var qualified = (QualifiedNameSyntax)node;
-                    return GetNodeName(qualified.Left, includeTypeParameters: false) + "." + GetNodeName(qualified.Right, includeTypeParameters: false);
-                case SyntaxKind.StructDeclaration:
-                    var structDecl = (StructDeclarationSyntax)node;
-                    name = structDecl.Identifier.ValueText;
-                    typeParameterList = structDecl.TypeParameterList;
-                    break;
-                default:
-                    Debug.Assert(false, "Unexpected node type " + node.Kind());
-                    return null;
+                    return $"{GetName(qualified.Left, options)}.{GetName(qualified.Right, options)}";
```

Comment:
Since this is a common path and there isn't any fancy formatting here, I think `string.Concat()` with only 3 arguments would be a better option than forcing the call to `string.Format()`.